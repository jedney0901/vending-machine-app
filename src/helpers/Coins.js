var Coins = [
  {
    amount: 10,
    type: '$0.1'
  },
  {
    amount: 20,
    type: '$0.2'
  },
  {
    amount: 50,
    type: '$0.5'
  },
  {
    amount: 100,
    type: '$1'
  },
  {
    amount: 200,
    type: '$2'
  }
]

export default Coins
