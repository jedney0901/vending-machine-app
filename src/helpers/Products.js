var Products = [
  {
    name: 'Caramel',
    cost: 250,
  },
  {
    name: 'Hazelnut',
    cost: 310,
  },
  {
    name: 'Organic Raw',
    cost: 200,
  }
]

export default Products
