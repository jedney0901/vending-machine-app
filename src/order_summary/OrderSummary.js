import React, {Component} from 'react';
import './OrderSummary.css'

class OrderSummary extends Component {

  render() {
    return (
      <div className="orderSummary bordered">
        Hello Worlds
      </div>
    );
  }
}

export default OrderSummary;