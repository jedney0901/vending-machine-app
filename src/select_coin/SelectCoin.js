import React, {Component} from 'react';
import {Button} from 'react-toolbox/lib/button/Button'

let btnStyles = {
  height: '40px',
  width: '60px',
  backgroundColor: '#262626',
  borderRadius: '10px',
  color: 'white',
  margin: '20px'
}

class SelectCoin extends Component {

  render() {
    return (
      <div>
        {
          this.props.coins.map((coin, i) =>
            <Button style={btnStyles} key={i} onClick={() => this.props.updateCredit(coin)}> {coin.type} </Button>
          )
        }
      </div>
    );
  }
}

export default SelectCoin;
