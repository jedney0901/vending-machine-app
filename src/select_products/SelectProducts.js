import React, {Component} from 'react';
import { Button } from 'react-toolbox/lib/button/Button'

class SelectProducts extends Component {

  render() {
    return (
      <div>
        {this.props.products.map((product, i) => <Button key={i} onClick={() => this.props.selectProducts(product)}> {product.name} </Button>)}
      </div>
    );
  }
}

export default SelectProducts;
