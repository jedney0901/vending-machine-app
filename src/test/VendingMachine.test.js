import React from 'react';
import ReactDOM from 'react-dom';
import {shallow} from 'enzyme';
import {Button} from 'react-toolbox/lib/button/Button'
import SelectCoin from '../select_coin/SelectCoin.js'
import SelectProducts from '../select_products/SelectProducts.js'
import OrderSummary from "../order_summary/OrderSummary"
import renderer from 'react-test-renderer';

// helpers
import Products from '../helpers/Products'
import Coins from '../helpers/Coins'

describe('Vending Machine Component', () => {
  const mockAddCredit = jest.fn();
  const mockSelectProducts = jest.fn();
  const mockResetOrder = jest.fn();
  const mockConfirmOrder = jest.fn();

  let initialState = {
    credit: 0,
    selected: [],
    confirm: 'no'
  }

  describe('SelectCoin component', () => {
    const component = shallow(
      <SelectCoin
        coins={Coins}
        updateCredit={mockAddCredit}
      />
    );

    it('renders and matches our snapshot', () => {
      const component = renderer.create(
        <SelectCoin
          coins={Coins}
          updateCredit={mockAddCredit}
        />
      );
    });
  });

  describe('SelectProduct component', () => {
    const component = shallow(
      <SelectProducts
        products={Products}
        updateCredit={mockSelectProducts}
      />
    );

    it('renders and matches our snapshot', () => {
      const component = renderer.create(
        <SelectProducts
          products={Products}
          updateCredit={mockSelectProducts}
        />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Confirm Order Button', () => {
    const component = shallow(
      <Button
        label='Purchase'
        onClick={mockConfirmOrder}
        className="confirmOrder"
      />
    );

    it('renders and matches our snapshot', () => {
      const component = renderer.create(
        <Button
          label='Purchase'
          onClick={mockConfirmOrder}
          className="confirmOrder"
        />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Reset Order Button', () => {
    const component = shallow(
      <Button
        coins={Coins}
        updateCredit={mockResetOrder}
        className="resetOrder"
      />
    );

    it('renders and matches our snapshot', () => {
      const component = renderer.create(
        <Button
          coins={Coins}
          updateCredit={mockResetOrder}
          className="resetOrder"
        />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Order Summary component', () => {
    const component = shallow(
      <OrderSummary
        products={Products}
        selectProducts={mockSelectProducts}
      />
    );

    it('renders and matches our snapshot', () => {
      const component = renderer.create(
        <OrderSummary
          products={Products}
          selectProducts={mockSelectProducts}
        />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});