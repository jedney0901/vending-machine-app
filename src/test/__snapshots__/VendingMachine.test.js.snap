// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vending Machine Component Confirm Order Button renders and matches our snapshot 1`] = `
<button
  className=" confirmOrder"
  data-react-toolbox="button"
  disabled={false}
  href={undefined}
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchStart={[Function]}
  type="button"
>
  Purchase
</button>
`;

exports[`Vending Machine Component Order Summary component renders and matches our snapshot 1`] = `
<div>
  <div />
</div>
`;

exports[`Vending Machine Component Reset Order Button renders and matches our snapshot 1`] = `
<button
  className=" resetOrder"
  coins={
    Array [
      Object {
        "amount": 10,
        "type": "$0.1",
      },
      Object {
        "amount": 20,
        "type": "$0.2",
      },
      Object {
        "amount": 50,
        "type": "$0.5",
      },
      Object {
        "amount": 100,
        "type": "$1",
      },
      Object {
        "amount": 200,
        "type": "$2",
      },
    ]
  }
  data-react-toolbox="button"
  disabled={false}
  href={undefined}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchStart={[Function]}
  type="button"
  updateCredit={[Function]}
/>
`;

exports[`Vending Machine Component SelectProduct component renders and matches our snapshot 1`] = `
<div>
  <button
    className=""
    data-react-toolbox="button"
    disabled={false}
    href={undefined}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "#F3C719",
        "borderRadius": "10px",
        "height": "40px",
        "margin": "20px",
        "width": "60px",
      }
    }
    type="button"
  >
     
    Caramel
     $
    2.5
     
  </button>
  <button
    className=""
    data-react-toolbox="button"
    disabled={false}
    href={undefined}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "#F3C719",
        "borderRadius": "10px",
        "height": "40px",
        "margin": "20px",
        "width": "60px",
      }
    }
    type="button"
  >
     
    Hazelnut
     $
    3.1
     
  </button>
  <button
    className=""
    data-react-toolbox="button"
    disabled={false}
    href={undefined}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "#F3C719",
        "borderRadius": "10px",
        "height": "40px",
        "margin": "20px",
        "width": "60px",
      }
    }
    type="button"
  >
     
    Organic Raw
     $
    2
     
  </button>
</div>
`;
